src/shared/components/events-list/index.js

```diff
@@ -2,134 +2,123 @@
 import DateBubble from '../date-bubble';
 import HR from '../hr';
 import { Grid, Cell } from '../grid';
-import classNames from 'classnames';
-import icons from '../icons/style.css';
 
 import TagsList from '../tags-list';
 import EventLinksList from '../event-links-list';
+import EventsTimelineTitle from '../events-timeline-title';
+import EventTitle from '../event-title';
 import { eventHref } from '../../util/event';
 import { splitEvents } from '../../util/split-events';
 
-export default class EventsList extends Component {
-  static propTypes = {
-    events: React.PropTypes.arrayOf(React.PropTypes.object).isRequired,
-    timeline: React.PropTypes.oneOf(['past', 'future', 'today']),
-  };
+const EventsList = ({
+  events,
+  timeline,
+}) => {
+  const relevantEvents =
+    timeline
+    ? splitEvents({
+      events,
+      timeline,
+      reverse: timeline === 'future',
+    })
+    : events;
 
-  render() {
-    const relevantEvents =
-      this.props.timeline
-      ? splitEvents(
-        this.props.events,
-        this.props.timeline,
-        { reverse: this.props.timeline === 'future' }
-      )
-      : this.props.events;
+  if (relevantEvents.length > 0) {
+    return (
+      <div className={styles.eventsListTimelineSection}>
+          <EventsTimelineTitle timeline={timeline} />
+        <ul className={styles.eventsList}>
+          {
+            relevantEvents.map((event) => (
+                <li key={`event_${event.id}`} className={styles.eventItem}>
+                  <Grid fit={false}>
+                    <Cell size={12}>
+                      <HR color="grey" customClassName=
+                        {styles.mobileHorizontalLine} />
+                      <DateBubble
+                          startDateTime={event.startDateTime}
+                          endDateTime={event.endDateTime}
+                          displayDateRange={(timeline === 'today' &&
+                            event.startDateTime.date !==
+                              event.endDateTime.date)}
+                      />
+                    </Cell>
+                    <Cell size={1} key="event_picture_mobile" hideOn="mobileSM">
+                      <EventImage imgPath={ imageAssetsEndpoint +
+                        (event.featureImageFilename ?
+                          event.featureImageFilename :
+                          'red-badger-event.jpg') } href={eventHref(event)} />
+                    </Cell>
+                    <Cell size={12} breakOn="mobile">
+                      <Grid fit={false}>
+                        <Cell size={8} key='event_description'
+                          breakOn="mobileS">
+                          <EventTitle eventTitle={event.title}
+                            eventHref={eventHref(event)} />
+                          <div className={styles.eventDescription}>
+                            {event.strapline}
+                          </div>
+                          {
```

Comment:
Can we do this in another function? Hard to read all in line. I think it's nice to keep logic and rendering apart.